desktop-widgets/downloadfromdivecomputer.cpp

```diff
@@ -116,7 +116,8 @@ void DownloadFromDCWidget::updateProgressBar()
 		ui.progressBar->setFormat("%p%");
 	}
 	ui.progressBar->setValue(lrint(progress_bar_fraction * 100));
-	free(last_text);
+	if (last_text)
```

Comment:
From the man page:
The free() function deallocates the memory allocation pointed to by ptr. If ptr is a NULL pointer, no operation is performed.
So passing NULL to free is valid. The only way this could crash was if something else free'ed last_text...
Based on that I'd say your commit doesn't look like the correct fix to me.